commit 056cfc16e3a7366845a7aecce32156749009032e
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2019-08-08T08:40:21+01:00 (9 weeks ago)
    
    joyent/node-vmapi-resolver#3 should return the tags of VM/backend

diff --git a/lib/resolver.js b/lib/resolver.js
index 38046f0..5d7809a 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -84,6 +84,20 @@ VmapiResolver.prototype.getBackends = function getBackends(cb) {
 			return;
 		}
 		Object.keys(vms).forEach(function (vm) {
+
+			var matched_tag;
+			if (vms[vm].tags.hasOwnProperty(vm_tag_name)) {
+				matched_tag = {
+				    'name': vm_tag_name,
+				    'value': vms[vm].tags[vm_tag_name]
+				};
+			} else {
+				log.warn({
+				    'vm_tags': vms[vm].tags,
+				    'vm_tag_name': vm_tag_name
+				}, 'tag name ambiguous; not returning');
+			}
+
 			vms[vm].nics.forEach(function (nic) {
 				/*
 				 * nic tags are matched by the caller-provided
@@ -92,12 +106,17 @@ VmapiResolver.prototype.getBackends = function getBackends(cb) {
 				if (nic_tag_regexp.test(nic.nic_tag)) {
 					backends.push({
 						'name': vms[vm].alias,
-						'address': nic.ip
+						'address': nic.ip,
+						'tag': matched_tag
 					});
 				}
 			});
 		});
-		log.info(backends, 'discovered backends');
+
+		log.info({
+		    'backends': backends,
+		}, 'discovered backends');
+
 		cb(null, backends);
 	});
 };
@@ -117,6 +136,11 @@ VmapiResolver.prototype.diffAndEmit = function diffAndEmit() {
 			'address': backend.address,
 			'port': self.vmr_backend_port
 		};
+
+		if (backend.tag) {
+			be['tag'] = backend.tag;
+		}
+
 		Object.keys(self.vmr_backends).forEach(function (key) {
 			old_backend = self.vmr_backends[key];
 			if (old_backend.address === be.address &&
